src/Engine/GameAction.java

```diff
@@ -109,6 +109,7 @@ public GameEventQueue getEvents(MapMaster gameMap)
     {
       GameEventQueue buildEvents = new GameEventQueue();
 
+      buildEvents.addAll(who.cos[0].revertActiveAbility(gameMap));
```

Comment:
Aww, that's mean. No lingering passive buffs if you pop a power and swap?

Reply:
It's that way for every implementation of tags I'm aware of.
Flip side: the power is no longer active, so you can immediately get charge for your opponent's counterattack.